### DISCUSSION TITLE
Reduce pressure on source by caching

### DISCUSSION THREAD

**Discussion by mahmoud-eskandari** [bluenviron/mediamtx#3502] [Questions]
### Question

Hi esteemed Mediamtx maintainers <3,

I have multiple proxy publisher servers that relay a stream from a single streaming server. I need assistance in finding the most efficient way to reduce the load on the streaming server.

streamer config:

```
paths:
  all_others:
     source: publisher
```

proxy publishers config:

```
paths:
  "~^air/(.+)$":
    source: rtsp://streamer-address:8554/air/$G1

```

I need the publisher servers to cache requests in either disk or memory to prevent call streamer server for every single requests otherwise publishers is useless.

**Do they operate this way by default?**

Thanks

**Comment by mahmoud-eskandari** [comment]
@aler9

**Comment by aler9** [comment]
Hello, the behavior you want is already enforced by default.

If some user requests URL `rtsp://proxy/air/mystream`, the proxy will reach down the streamer server and start downloading stream `rtsp://streamer-address:8554/air/mystream`.

If another user request the same URL, the proxy won't download the stream twice from streamer server, but will use the existing stream.

[a comment by an automated account is not shown]
